core/templates/pages/learner-dashboard-page/new-home-tab.component.spec.ts

```diff
@@ -0,0 +1,572 @@
+        thumbnail_filename: 'image.svg',
+        thumbnail_bg_color: '#F8BF74',
+        story_is_published: true,
+        completed_node_titles: ['Chapter 1'],
+        url_fragment: 'story-title',
+        all_node_dicts: [nodeDict1]
+      },
+      {
+        id: '1',
+        title: 'Story Title',
+        description: 'Story Description',
+        node_titles: ['Chapter 1'],
+        thumbnail_filename: 'image.svg',
+        thumbnail_bg_color: '#F8BF74',
+        story_is_published: true,
+        completed_node_titles: [],
+        url_fragment: 'story-title',
+        all_node_dicts: [nodeDict2]
+      }],
+      url_fragment: 'topic-name',
+      subtopics: [subtopic],
+      degrees_of_mastery: {
+        skill_id_1: 0.5,
+        skill_id_2: 0.3
+      },
+      skill_descriptions: {
+        skill_id_1: 'Skill Description 1',
+        skill_id_2: 'Skill Description 2'
+      }
+    };
+    component.currentGoals = [LearnerTopicSummary.createFromBackendDict(
+      learnerTopicSummaryBackendDict1)];
+    component.goalTopics = [LearnerTopicSummary.createFromBackendDict(
+      learnerTopicSummaryBackendDict1)];
+    component.partiallyLearntTopicsList = [];
+    component.storyInProgress = [];
+    component.storyInRecommended = [];
+    component.incompleteExplorationsList = [];
+    component.explorationPlaylist = [];
+    component.collectionPlaylist = [];
+    component.completedExplorationsList = [];
+    component.untrackedTopics = {};
+    component.storyIdToLearnerGroupsTitleMap = new Map([
+      ['0', ['learner Group Title']]]);
+    component.username = 'username';
+    fixture.detectChanges();
+  });
+
+  it('should get the correct width in mobile view', () => {
+    expect(component.storyInProgress.length).toBe(1);
+  });
+
+  it('should check the number of completed exploration list', () => {
+    const completedLearnerExplorationSummaryDict = {
+      last_updated_msec: 1591296737470.528,
+      community_owned: false,
+      objective: 'Test Objective',
+      id: '44LKoKLlIbGe',
+      num_views: 0,
+      thumbnail_icon_url: '/subjects/Algebra.svg',
+      human_readable_contributors_summary: {},
+      language_code: 'en',
+      thumbnail_bg_color: '#cc4b00',
+      created_on_msec: 1591296635736.666,
+      ratings: {
+        1: 0,
+        2: 0,
+        3: 0,
+        4: 0,
+        5: 0
+      },
+      status: 'public',
+      tags: [],
+      activity_type: 'exploration',
+      category: 'Algebra',
+      title: 'Test Title'
+    };
+    let completedSummary = LearnerExplorationSummary.createFromBackendDict(
+      completedLearnerExplorationSummaryDict);
+    component.completedExplorationsList = [completedSummary];
+    const completedExpIds = ['44LKoKLlIbGe'];
+    component.ngOnInit();
+    expect(component.completedExpIds).toEqual(completedExpIds);
+  });
+  it('should populate exploration ID count in learner exploration summary',
+    () => {
+      const exploration1 = {
```

Comment:
What does the "1" here signify? Avoid appending numbers to variables unless they convey some meaning. There is only one exploration in this test, so just "exploration" might do.

Reply:
Ditto below in this file.